### PR TITLE
Vuln flavor refactor to sub domains

### PR DESCRIPTION
**Pull request by amirbenun**
### Summary of your changes
Refactor the vulnerability flavor and separating its functionality to sub domains and adding unit tests.

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: flavors/publisher_test.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+		{
+			name:              "Publish events on threshold reached",
+			interval:          time.Minute,
+			threshold:         1,
+			ctxTimeout:        200 * time.Millisecond,
+			eventCount:        2,
+			expectedEventSize: []int{1},
+		},
+		{
+			name:              "Publish events on threshold reached",
```
Comment (olegsu) [inline flavors/publisher_test.go:52]: Why have two tests with the same purpose? 
Either update the name or delete the test

File: flavors/publisher_test.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+		{
+			name:              "Publish events on context done",
+			interval:          time.Minute,
+			threshold:         5,
+			ctxTimeout:        100 * time.Millisecond,
+			eventCount:        3,
+			expectedEventSize: []int{3},
+		},
+		{
+			name:              "Publish 0 events on context done",
```
Comment (olegsu) [inline flavors/publisher_test.go:101]: Also here

File: flavors/client.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+	return pipeline.ConnectWith(beat.ClientConfig{
+		Processing: beat.ProcessingConfig{
+			Processor: procs,
+		},
+	})
+}
+
+// ConfigureProcessors configure processors to be used by the beat
+func ConfigureProcessors(processorsList processors.PluginConfig) (procs *processors.Processors, err error) {
```
Comment (olegsu) [inline flavors/client.go:39]: nit: can be private

**Review by olegsu — APPROVED** [review]

File: flavors/publisher_test.go (hunk 2 above)
Comment (amirbenun) [reply flavors/publisher_test.go:101]: Not exactly the same in that case

File: flavors/publisher.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+)
+
+type client interface {
+	PublishAll([]beat.Event)
+}
+
+type Publisher struct {
+	log      *logp.Logger
+	interval time.Duration
+	treshold int
```
Comment (orestisfl) [inline flavors/publisher.go:35]:
```suggestion
	threshold int
```

File: flavors/publisher.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	client   client
+}
+
+func NewPublisher(log *logp.Logger, interval time.Duration, treshold int, client client) *Publisher {
```
Comment (orestisfl) [inline flavors/publisher.go:39]:
```suggestion
func NewPublisher(log *logp.Logger, interval time.Duration, threshold int, client client) *Publisher {
```

File: flavors/publisher.go [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+	}
+}
+
+func (p *Publisher) HandleEvents(ctx context.Context, ch chan []beat.Event) {
+	var eventsToSend []beat.Event
+	flushTicker := time.NewTicker(p.interval)
+	for {
+		select {
+		case <-ctx.Done():
+			p.log.Warn("Publisher context is done")
```
Comment (orestisfl) [inline flavors/publisher.go:54]:
```suggestion
			p.log.Warnf("Publisher context is done: %v", ctx.Err())
```

slightly more informative

File: flavors/publisher_test.go [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+			eventCount:        4,
+			expectedEventSize: []int{6},
+			closeChannel:      true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			log := testhelper.NewLogger(t)
+			goleak.VerifyNone(t, goleak.IgnoreCurrent())
```
Comment (orestisfl) [inline flavors/publisher_test.go:138]:
```suggestion
			defer goleak.VerifyNone(t, goleak.IgnoreCurrent())
```

File: flavors/repeater.go [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+func (r *Repeater) Run(ctx context.Context, fn RepeaterFunc) error {
+	r.log.Warn("Repeater ticker running for period ", r.interval)
+	ticker := time.NewTicker(r.interval)
+	immediate := time.NewTimer(0)
+
+	for {
+		select {
+		case <-ctx.Done():
+			r.log.Warn("Repeater context is done")
```
Comment (orestisfl) [inline flavors/repeater.go:49]:
```suggestion
			r.log.Warnf("Repeater context is done: %v", ctx.Err())
```

**Pull request merged**